src/Quaternion.h

```diff
@@ -30,21 +30,42 @@ class Quaternion {
 		y = axis.y * sinHalfAng;
 		z = axis.z * sinHalfAng;
 	}
-	// from axis, assume angle == PI
-	// optimized fast path using sin(PI/2) = 1
-	Quaternion(vector3<T> axis)
+	// Create quaternion from normalized direction vectors.
+	// This creates a quaternion representing the rotation from the second
+	// unit direction vector to the first unit direction vector.
+	Quaternion(vector3<T> to, vector3<T> from)
```

Comment:
Any particular reason we're passing by-value instead of by-const-ref here?

Reply:
This particular constructor is circa 2021... that'd be why. I'll fix that.

Reply:
...?  this has been part of C++ since .. forever. 

Since C++03 (more so since C++11 with move semantics) there are some performance benefits passing a complex type by value if you're going to make a copy of it anyway, but apart from that you would always use (const) references for parameters.

C++ is absolutely horrendous in its complicated minutely different details; there's a reason I generally prefer working in C. For example, [there's a book](https://www.amazon.com/Initialization-Story-Through-Options-Related/dp/B0BW38DDBK) on the different ways to initialise variables in C++; it runs to 295 pages or so (no joke). The equivalent book for C would be 1 page. 2 at most.

Reply:
Yeah but not all of it was written by sturnclaw, it's just being resurrected. Please don't be so harsh in review, some of our contributors are learning the language(s) used.

Reply:
My apologies, I asked whether there was a reason and the answer was just "2021".. that didn't make sense to me. The rest of my comment was basically acknowledging that C++ is an incredibly complex beast which few can hope to be reasonable at, let alone master (at best I myself fall into the former category; certainly not into the latter).

Reply:
..and, not a real excuse but background, I've recently been doing a lot of code reviews at work which is _extremely_ strict and I'm obviously not managing to tone it down enough for hobby-level here. Again, my apologies. No offence or overt criticality meant.  There should probably have been a smiley somewhere in my comment to make it more obvious that it wasn't meant that harshly.

Reply:
No worries, just didn't want critique to distract from a good solid review 👍

Reply:
I would like to add that in the disassembler (x86) I did not see `Quaternion` constructors at all, apparently they are all inlined and optimized.

Reply:
> I would like to add that in the disassembler (x86) I did not see `Quaternion` constructors at all, apparently they are all inlined and optimized.

Quaternion was used in an intermediate stage in the GunManager, but is not currently used (as it's more efficient to avoid a round-trip through a Quaternion for the code's specific needs).